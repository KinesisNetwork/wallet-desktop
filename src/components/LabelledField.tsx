import * as copy from 'copy-to-clipboard'
import * as React from 'react'

export interface Props {
  label: string
  value: string
  icon?: string
  isLoading?: boolean
  isCompact?: boolean
  isClipped?: boolean
  addon?: React.ReactNode
  isCopyable?: boolean
}

const clippedStyle = (isClipped: boolean | undefined) => isClipped ? { maxWidth: '35ch' } : {}

export const LabelledField: React.SFC<Props> = (props) => (
  <div className='field'>
    <label className='label is-small'>{props.label}</label>
    <div className='field is-grouped'>
      <div className={`control is-expanded ${props.isLoading && 'is-loading'}`} style={clippedStyle(props.isClipped)}>
        {
          props.isCopyable &&
          <button
            className='button is-text is-paddingless'
            onClick={() => copy(props.value)}
            title='Click to copy'
          >
            {props.value}
          </button>
        }
        {
          !props.isCopyable &&
          <p className='input is-static'>{props.value}</p>
        }
      </div>
      {props.addon && <div className='control'>{props.addon}</div>}
    </div>
  </div>
)

LabelledField.displayName = 'LabelledField'

export const HorizontalLabelledField: React.SFC<Props> = (props) => (
  <div className={`field is-horizontal ${props.isCompact ? 'is-marginless' : ''}`}>
    <div className='field-label is-small'>
      <label className='label'>{props.label}</label>
    </div>
    <div className='field-body'>
      <div className='field is-grouped'>
        <div className={`control is-expanded ${props.isLoading && 'is-loading'}`} style={clippedStyle(props.isClipped)}>
          <p className='input is-static is-small is-block is-clipped' style={{ textOverflow: 'ellipsis' }} >
            {props.value}
          </p>
        </div>
        {props.addon && <div className='control'>{props.addon}</div>}
      </div>
    </div>
  </div>
)

HorizontalLabelledField.displayName = 'HorizontalLabelledField'
