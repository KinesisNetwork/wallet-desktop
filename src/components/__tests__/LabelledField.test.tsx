import * as Enzyme from 'enzyme'
import { shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16'
import * as React from 'react'
import { LabelledField } from '../LabelledField'

Enzyme.configure({ adapter: new Adapter() })

describe('LabelledField', () => {

  describe('should render', () => {
    it('correctly', () => {
      const wrapper: Enzyme.ShallowWrapper = shallow(<LabelledField label='Test' value='Test value' />)

      expect(wrapper).toMatchSnapshot()
    })

    it('a label', () => {
      const wrapper: Enzyme.ShallowWrapper = shallow(<LabelledField label='Test' value='Test value' />)

      const label: Enzyme.ShallowWrapper = wrapper.find('label')

      expect(label).toHaveLength(1)
      expect(label.text()).toEqual('Test')
    })

    it('a parapgraph when the isCopyable property is set to false or does not exist', () => {
      const wrapper: Enzyme.ShallowWrapper = shallow(<LabelledField label='Test' value='Test value' />)

      const paragraph: Enzyme.ShallowWrapper = wrapper.find('p')
      const button: Enzyme.ShallowWrapper = wrapper.find('button')

      expect(paragraph).toHaveLength(1)
      expect(paragraph.text()).toEqual('Test value')
      expect(button).toHaveLength(0)
    })

    it('a button element when isCopyable is true', () => {
      const wrapper: Enzyme.ShallowWrapper = shallow(<LabelledField label='Test' value='Test value' isCopyable={true} />)

      const paragraph: Enzyme.ShallowWrapper = wrapper.find('p')
      const button: Enzyme.ShallowWrapper = wrapper.find('button')

      expect(paragraph).toHaveLength(0)
      expect(button).toHaveLength(1)
      expect(button.text()).toEqual('Test value')
    })
  })
})
