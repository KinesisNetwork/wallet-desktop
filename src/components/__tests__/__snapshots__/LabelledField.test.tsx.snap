// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledField should render correctly 1`] = `
<div
  className="field"
>
  <label
    className="label is-small"
  >
    Test
  </label>
  <div
    className="field is-grouped"
  >
    <div
      className="control is-expanded undefined"
      style={Object {}}
    >
      <p
        className="input is-static"
      >
        Test value
      </p>
    </div>
  </div>
</div>
`;
